File: commonutils/src/main/java/net/maxsmr/commonutils/logger/Slf4Logger.java
package net.maxsmr.commonutils.logger;

import android.support.annotation.NonNull;


import net.maxsmr.commonutils.logger.base.BaseSimpleLogger;

import org.slf4j.Logger;

public class Slf4Logger extends BaseSimpleLogger {

    @NonNull
    private final Logger logger;

    public Slf4Logger(@NonNull Logger logger) {
        this.logger = logger;
    }

    @Override
    public void info(String message) {
        if (isLoggingEnabled()) {
            logger.info(message);
        }
    }

    @Override
    public void debug(String message) {
        if (isLoggingEnabled()) {
            logger.debug(message);
        }
    }

    @Override
    public void warn(String message) {
        if (isLoggingEnabled()) {
            logger.warn(message);
        }
    }

    @Override
    public void warn(Throwable exception) {
        if (isLoggingEnabled()) {
            logger.warn("", exception);
        }
    }

    @Override
    public void warn(String message, Throwable exception) {
        if (isLoggingEnabled()) {
            logger.warn(message, exception);
        }
    }

    @Override
    public void error(String message) {
        if (isLoggingEnabled()) {
            logger.error(message);
        }
    }

    @Override
    public void error(Throwable exception) {
        if (isLoggingEnabled()) {
            logger.error("", exception);
        }
    }

    @Override
    public void error(String message, Throwable exception) {
        if (isLoggingEnabled()) {
            logger.error(message, exception);
        }
    }
}

File: testapp/src/main/kotlin/net/maxsmr/testapp/logger/Slf4Logger.kt
package net.maxsmr.testapp.logger

import net.maxsmr.commonutils.logger.BaseLogger
import org.slf4j.Logger

class Slf4Logger(private val logger: Logger) : BaseLogger(logger.toString()) {
    
    override fun v(message: String) {
        if (isLoggingEnabled) {
            logger.trace(message)
        }
    }

    override fun v(exception: Throwable) {
        if (isLoggingEnabled) {
            logger.trace(exception.message)
        }
    }

    override fun v(message: String, exception: Throwable?) {
        if (isLoggingEnabled) {
            logger.trace(message, exception)
        }
    }

    override fun d(message: String) {
        if (isLoggingEnabled) {
            logger.debug(message)
        }
    }

    override fun d(exception: Throwable) {
        if (isLoggingEnabled) {
            logger.debug(exception.message)
        }
    }

    override fun d(message: String, exception: Throwable?) {
        if (isLoggingEnabled) {
            logger.debug(message, exception)
        }
    }

    override fun i(message: String) {
        if (isLoggingEnabled) {
            logger.info(message)
        }
    }

    override fun i(exception: Throwable) {
        if (isLoggingEnabled) {
            logger.info(exception.message)
        }
    }

    override fun i(message: String, exception: Throwable?) {
        if (isLoggingEnabled) {
            logger.info(message, exception)
        }
    }

    override fun w(message: String) {
        if (isLoggingEnabled) {
            logger.warn(message)
        }
    }

    override fun w(exception: Throwable) {
        if (isLoggingEnabled) {
            logger.warn(exception.message)
        }
    }

    override fun w(message: String, exception: Throwable?) {
        if (isLoggingEnabled) {
            logger.warn(message, exception)
        }
    }

    override fun e(message: String) {
        if (isLoggingEnabled) {
            logger.error(message)
        }
    }

    override fun e(exception: Throwable) {
        if (isLoggingEnabled) {
            logger.error(exception.message)
        }
    }

    override fun e(message: String, exception: Throwable?) {
        if (isLoggingEnabled) {
            logger.error(message, exception)
        }
    }

    override fun wtf(message: String) {
        e(message)
    }

    override fun wtf(exception: Throwable) {
        e(exception)
    }

    override fun wtf(message: String, exception: Throwable?) {
        e(message, exception)
    }
}
